beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -750,3 +757,371 @@ func (bs *Server) checkSync(ctx context.Context, w http.ResponseWriter) bool {
+		randaoReveal = primitives.PointAtInfinity
+	} else {
+		rr, err := hexutil.Decode(rawRandaoReveal)
+		if err != nil {
+			errJson := &http2.DefaultErrorJson{
+				Message: errors.Wrap(err, "unable to decode randao reveal").Error(),
+				Code:    http.StatusBadRequest,
+			}
+			http2.WriteError(w, errJson)
+			return
+		}
+		randaoReveal = rr
+	}
+	if len(randaoReveal) != fieldparams.BLSSignatureLength {
+		errJson := &http2.DefaultErrorJson{
+			Message: fmt.Sprintf("a valid randao reveal is required as a query parameter: received length %d but wanted length %d", len(randaoReveal), fieldparams.BLSSignatureLength),
+			Code:    http.StatusBadRequest,
+		}
+		http2.WriteError(w, errJson)
+		return
+	}
+	var graffiti []byte
+	if rawGraffiti != "" {
+		g, err := hexutil.Decode(rawGraffiti)
+		if err != nil {
+			errJson := &http2.DefaultErrorJson{
+				Message: errors.Wrap(err, "unable to decode graffiti").Error(),
+				Code:    http.StatusBadRequest,
+			}
+			http2.WriteError(w, errJson)
+			return
+		}
+		graffiti = g
+	}
+
+	produceBlockV3(bs, w, r, &eth.BlockRequest{
+		Slot:         primitives.Slot(slot),
+		RandaoReveal: randaoReveal,
+		Graffiti:     graffiti,
+		SkipMevBoost: false,
+	})
+
+}
+
+func produceBlockV3(bs *Server, w http.ResponseWriter, r *http.Request, v1alpha1req *eth.BlockRequest) {
+	isSSZ, err := http2.SszRequested(r)
+	if err != nil {
+		log.WithError(err).Error("verifying ssz request failed, defaulting to non ssz.")
+		isSSZ = false
+	}
+	validate := validator.New()
+	v1alpha1resp, err := bs.V1Alpha1ValidatorServer.GetBeaconBlock(r.Context(), v1alpha1req)
+	if err != nil {
+		http2.HandleError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	phase0Block, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_Phase0)
+	if ok {
+		handleProducePhase0V3(isSSZ, phase0Block, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	altairBlock, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_Altair)
+	if ok {
+		handleProduceAltairV3(isSSZ, altairBlock, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	optimistic, err := bs.OptimisticModeFetcher.IsOptimistic(r.Context())
+	if err != nil {
+		http2.HandleError(w, errors.Wrap(err, "Could not determine if the node is a optimistic node").Error(), http.StatusInternalServerError)
+		return
+	}
+	if optimistic {
+		http2.HandleError(w, "The node is currently optimistic and cannot serve validators", http.StatusInternalServerError)
+		return
+	}
+	blindedBellatrixBlock, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_BlindedBellatrix)
+	if ok {
+		handleProduceBlindedBellatrixV3(isSSZ, blindedBellatrixBlock, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
+	}
+	bellatrixBlock, ok := v1alpha1resp.Block.(*eth.GenericBeaconBlock_Bellatrix)
+	if ok {
+		handleProduceBellatrixV3(isSSZ, bellatrixBlock, w, validate, v1alpha1resp.IsBlinded, v1alpha1resp.PayloadValue)
+		return
```

Comment:
this can be inside a conditional
```
if v1alpha1resp.IsBlinded {
	// check blinded
} else {
	// check unblinded
}
```
same for capella and deneb

Reply:
what do you mean by this shouldn't it match the type check?

Reply:
What I mean is that it doesn't make sense to type check for a blinded block when `v1alpha1resp.IsBlinded == false` and vice versa.